compiler/rustc_lint/src/unqualified_local_imports.rs

```diff
@@ -47,17 +46,12 @@ declare_lint_pass!(UnqualifiedLocalImports => [UNQUALIFIED_LOCAL_IMPORTS]);
 impl<'tcx> LateLintPass<'tcx> for UnqualifiedLocalImports {
     fn check_item(&mut self, cx: &LateContext<'tcx>, item: &'tcx hir::Item<'tcx>) {
         let hir::ItemKind::Use(path, _kind) = item.kind else { return };
-        // `path` has three resolutions for the type, module, value namespaces.
-        // Check if any of them qualifies: local crate, and not a macro.
-        // (Macros can't be imported any other way so we don't complain about them.)
-        let is_local_import = |res: &Res| {
-            matches!(
-                res,
-                hir::def::Res::Def(def_kind, def_id)
-                    if def_id.is_local() && !matches!(def_kind, DefKind::Macro(_)),
-            )
-        };
-        if !path.res.iter().any(is_local_import) {
+        // Check the type namespace resolution for a local crate.
+        let is_local_import = matches!(
+            path.res.value_ns,
```

Comment:
The comment says type namespace, the actual logic uses value namespace, and the previous logic considered both.

Reply:
That was a strange one, because the tests pass so long as either the type namespace *or* the value namespace is checked. I'll change it to check both.